code/modules/economy/store/store_credits.dm

```diff
@@ -23,49 +31,52 @@
 	return (get_available_credits() >= cost)
 
 
-
-
 //Deducts credits
-/obj/machinery/store/proc/occupant_pay_credits(var/cost, var/safety = TRUE)
-	if (safety && (!occupant || !occupant_can_afford()))
+/obj/machinery/store/proc/occupant_pay_credits(left_to_pay)
+	if(!occupant || !occupant_can_afford())
 		return FALSE
 
-
-	var/pay_amount
-	if (chip)
-		pay_amount = min(cost, chip.worth)
-		if (chip.pay_into(pay_amount, src))
-			cost -= pay_amount
-
-	if (cost <= 0)
-		return TRUE
-
-	//Alright now we pay whatever is leftover from the rig account
-
-
-
-	if (cost > 0)
-		if (occupant && occupant.wearing_rig)
-			pay_amount = min(occupant.wearing_rig.get_account_balance(), cost)
+	if(chip)
+		if(chip.worth <= left_to_pay)
+			left_to_pay -= chip.worth
+			chip.worth = 0
+			chip.update_icon()
+		else
+			chip.worth -= left_to_pay
+			chip.update_icon()
+			return TRUE
+
+	if(occupant && occupant.wearing_rig)
+		var/pay_amount = min(occupant.wearing_rig.get_account_balance(), left_to_pay)
 		var/obj/item/weapon/rig/R = occupant.wearing_rig
 		R.charge_to_rig_account(src, "Store Purchase", machine_id, -pay_amount)
-		cost -= pay_amount
+		left_to_pay -= pay_amount
 
+	if(left_to_pay)
+		for(var/i in deposit_box.contents)
+			if(istype(i, /obj/item/weapon/spacecash))
+				var/obj/item/weapon/spacecash/S = i
+				if(S.worth <= left_to_pay)
+					left_to_pay -= S.worth
+					qdel(S)
+					deposit_box.update_ui_data()
+				else
+					S.worth -= left_to_pay
+					S.update_icon()
+					deposit_box.update_ui_data()
+					return TRUE
 
-	if (cost <= 0)
-		return TRUE
+	var/datum/money_account/A = occupant?.get_account()
+	if(A)
+		A.money -= left_to_pay
```

Comment:
There are no safety checks here meaning our account balance can become lower than 0

Reply:
We get here only if occupant_can_afford() returned TRUE, but no money sources other than this one left,
hence check here is redundant, as there is no cases for 'not enough money' at this point.